src/plugins/fcrypt/fcrypt.c

```diff
@@ -0,0 +1,387 @@
+/**
+ * @file
+ *
+ * @brief filter plugin providing cryptographic operations
+ *
+ * @copyright BSD License (see doc/COPYING or http://www.libelektra.org)
+ *
+ */
+
+#ifndef HAVE_KDBCONFIG
+#include "kdbconfig.h"
+#endif
+#include "fcrypt.h"
+#include <errno.h>
+#include <fcntl.h>
+#include <gpg.h>
+#include <kdb.h>
+#include <kdberrors.h>
+#include <kdbtypes.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+enum fcryptGetState
+{
+	PREGETSTORAGE = 0,
+	POSTGETSTORAGE = 1
+};
+
+struct _fcryptState
+{
+	enum fcryptGetState getState;
+};
+typedef struct _fcryptState fcryptState;
+
+/**
+ * @brief Allocates a new string holding the name of the temporary file.
+ * @param file holds the path to the original file
+ * @returns an allocated string holding the name of the encrypted file. Must be freed by the caller.
+ */
+static char * getTemporaryFileName (const char * file)
+{
+	const size_t newFileAllocated = strlen (file) + 7;
+	char * newFile = elektraMalloc (newFileAllocated);
+	if (!newFile) return NULL;
+	snprintf (newFile, newFileAllocated, "%sXXXXXX", file);
+	mktemp (newFile);
```

Comment:
This would lead to the annoying "warning: the use of mktemp is dangerous, better use mkstemp or mkdtemp" (even if it will not add much to security here, i.e. only if the config file is in a group/world writeable directory)

The returned file handle is useful for overwriting the file with zeros anyway? (Or does gpg unlink existing files first?)